tutorials/backend/hasura-v3-ts-connector/tutorial-site/content/setup-connector.md

```diff
@@ -0,0 +1,607 @@
+just defined.
+
+Now we can put the schema response together:
+
+```typescript
+async function get_schema(configuration: RawConfiguration): Promise<SchemaResponse> {
+  let scalar_types: { [k: string]: ScalarType } = {
+    'any': {
+      aggregate_functions: {},
+      comparison_operators: {},
+      update_operators: {},
+    }
+  };
+
+  let object_types: { [k: string]: ObjectType } = {};
+
+  for (const table of configuration.tables) {
+    let fields: { [k: string]: ObjectField } = {};
+
+    for (const columnName in table.columns) {
+      fields[columnName] = {
+        type: {
+          type: 'named',
+          name: 'any'
+        }
+      };
+    }
+
+    object_types[table.tableName] = {
+      fields
+    };
+  }
+
+  let collections: CollectionInfo[] = configuration.tables.map((table) => {
+    return {
+      arguments: {},
+      name: table.tableName,
+      deletable: false,
+      foreign_keys: {},
+      uniqueness_constraints: {},
+      type: table.tableName,
+    };
+  });
+
+  return {
+    functions: [],
+    procedures: [],
+    collections,
+    object_types,
+    scalar_types,
+  };
+}
+```
+
+Notice that we don't define `functions` or `procedures`, but we'll cover those features later in the series.
+
+## Testing
+
+So we have one more function to define, which is the query function, but before we do, let's talk about tests. The [NDC
+specification repository](https://github.com/hasura/ndc-spec/) provides a 
+[test runner executable](https://github.com/hasura/ndc-spec/tree/main/ndc-test) called `ndc-test`, which can be used to 
+implement a test suite for a connector. 
+
+We can also use ndc-test to run some automatic tests and validate the work we've done so far. Let's compile and run 
+our connector, and then use the test runner with the running connector.
+
+Here I have a configuration.json file which I can use to run the connector against my sample database.
+
+First let's run the connector.
+
+```shell
+npm run build && node dist/index.js serve --configuration configuration.json
+```
+
+[//]: # (TODO - document the test runner better in the spec repo)
+Now, let's run the tests. (You will need to have the 
+[ndc test runner](https://github.com/hasura/graphql-engine-mono/pull/10626#pullrequestreview-1826948556) installed on 
+your machine.)
+
+```shell
+ndc-test test --endpoint http://localhost:8100
+```
+
+OR
+
+```shell
+cargo run --bin ndc-test -- test --endpoint http://localhost:8100
+````
+
+Of course, we expected it to fail, but we can already see that our schema response is good.
+
+Let's modify our query function to print out the request it receives, and this will give us a goal to work towards.
+
+```typescript
+async function query(configuration: RawConfiguration, state: State, request: QueryRequest): Promise<QueryResponse> {
+  console.log(JSON.stringify(request, null, 2));
+  throw new Error("Function not implemented.");
+}
+```
+
+Let's recompile and restart the connector, and run the tests again.
```

Comment:
Would it make sense to have `nodemon` running and explained in the setup so a user doesn't have to recompile each time they make changes?

If so, we could add it under a `watch` script so the DX is similar to what they'd experience using the new CLI.

Reply:
done